File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/jetstream/os/ObjectStoreTest.java
package io.micronaut.nats.docs.jetstream.os;

import io.micronaut.context.annotation.Property;
import io.micronaut.nats.testcontainers.Nats;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import io.nats.client.JetStreamApiException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.security.NoSuchAlgorithmException;
import java.util.Map;

@MicronautTest
@Property(name = "spec.name", value = "ObjectStoreTest")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class ObjectStoreTest implements TestPropertyProvider {

    @Test
    void simpleTest(ObjectStoreHolder objectStoreHolder) throws JetStreamApiException, IOException, NoSuchAlgorithmException {
        objectStoreHolder.put("test", new ByteArrayInputStream("myvalue".getBytes()));

        Assertions.assertNotNull(objectStoreHolder.store.getInfo("test"));
    }

    @Override
    public Map<String, String> getProperties() {
        return Nats.getProperties();
    }
}

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/jetstream/os/ObjectStoreTest.py
from typing import Annotated

from jakarta.inject import Inject
from java.io import ByteArrayInputStream
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .ObjectStoreHolder import ObjectStoreHolder


@Property(name="spec.name", value="ObjectStoreTest")
@MicronautTest(environments=["nats"])
class ObjectStoreTest:
    object_store_holder: Annotated[ObjectStoreHolder, Inject]

    @Test
    def simple_test(self):
        self.object_store_holder.put("test", ByteArrayInputStream("myvalue".encode()))

        assert self.object_store_holder.store.getInfo("test") is not None
